### DISCUSSION TITLE
Help about adding Javasript to the footer.

### DISCUSSION THREAD

**Discussion by mendax1234** [squidfunk/mkdocs-material#5944] [Q&A]
Hi, l want to add a widget to my website using Javascript. l hope this widget can be added at the bottom of one page and above the footer, just like where the `feedback` is on one page. First, l try to add `main.html`, however, l find that the widget is always below the footer. It looks like below:
<img width="960" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/62934294/6443ec93-3e9a-4747-8160-353faf60aa72">

Then, l try to add this widget by changing the `analytics.html` since l think the position of `feedback about whether the page is helpful or not` is what l want. However, after changing that, the result is the same as above mentioned. Besides, l also want to use `google` as my analytics provider, so l didn't try using a `custom.html` in my analytics.

Below is the Javascript code l use to generate the widget:
```
    <script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=ffffff&w=200&t=tt&d=Z-7P-c__jyKWkEFWv6kpimyIUKQSYn8vWo0XVWtk0gE'></script>
```

Thanks for any help!

**Comment by mendax1234** [comment]
Hi, I think l have successfully achieved what l want by using adding this script into `main.html`
```
{% extends "base.html" %}

{% block footer %}
  <script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=080808&w=300&t=tt&d=Z-7P-c__jyKWkEFWv6kpimyIUKQSYn8vWo0XVWtk0gE&co=ffffff&cmo=3acc3a&cmn=ff5353&ct=808080'></script>
  {{ super() }}
{% endblock %}
```

l think that adding a script in the `footer` is quite strange, but l find that only in this way can l achieve what l want. Letting the widget display at the bottom of the page but above the footer. However, l am curious about whether there are any other methods to achieve this.

<img width="946" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/62934294/644e19a9-6020-4d33-acaf-98706d7329bb">

Besides, l still think this way may waste room. So, l wonder if l can achieve the below result.
<img width="938" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/62934294/6bfe2d5d-c110-4789-845f-d0c3f05e5763">

l  guess that to achieve this, l should spend some time in modifying the `footer.html`. Are there any good suggestions?

**Reply by squidfunk** [reply]
You can append it to the [`copyright`](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/copyright.html) partial or prepend it to the [`social`](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/social.html) partial.

**Reply by mendax1234** [reply]
Hi, thanks for your reply! After doing this, l find that the copyright, the small widget, and the social icon are not on the same line. Is there any way that l can use it to center the social icons? Or center the small widget? Thanks!
